kubevirt-disk-uploader-task.yaml

```diff
@@ -0,0 +1,95 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubevirt-disk-uploader-tekton
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubevirt-disk-uploader-tekton
+rules:
+- apiGroups: ["kubevirt.io"]
+  resources: ["virtualmachines"]
+  verbs: ["get", "create", "patch"]
+- apiGroups: ["export.kubevirt.io"]
+  resources: ["virtualmachineexports"]
+  verbs: ["get", "create"]
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "create"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kubevirt-disk-uploader-tekton
+subjects:
+- kind: ServiceAccount
+  name: kubevirt-disk-uploader-tekton
+roleRef:
+  kind: Role
+  name: kubevirt-disk-uploader-tekton
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kubevirt-disk-uploader-credentials-tekton
+type: Opaque
+data:
+  registryUsername: "<REGISTRY_USERNAME>"
+  registryPassword: "<REGISTRY_PASSWORD>"
+  registryHostname: "<REGISTRY_HOSTNAME>"
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kubevirt-disk-uploader-task
+spec:
+  params:
+  - name: VM_NAME
+    description: The name of the virtual machine
+    type: string
+  - name: VOLUME_NAME
+    description: The volume name of the virtual machine
+    type: string
+  - name: CONTAINER_DISK_NAME
+    description: The name of the new image
+    type: string
+  - name: ENABLE_VIRT_SYSPREP
+    description: Enable or disable preparation of disk image
+    type: string
+  steps:
+  - name: kubevirt-disk-uploader-step
+    image: quay.io/boukhano/kubevirt-disk-uploader:latest
+    env:
+    - name: REGISTRY_USERNAME
+      valueFrom:
+        secretKeyRef:
+          name: kubevirt-disk-uploader-credentials-tekton
+          key: registryUsername
+    - name: REGISTRY_PASSWORD
+      valueFrom:
+        secretKeyRef:
+          name: kubevirt-disk-uploader-credentials-tekton
+          key: registryPassword
+    - name: REGISTRY_HOST
+      valueFrom:
+        secretKeyRef:
+          name: kubevirt-disk-uploader-credentials-tekton
+          key: registryHostname
+    command: ["/usr/local/bin/kubevirt-disk-uploader"]
+    args:
+    - "--vmname"
+    - $(params.VM_NAME)
+    - "--volumename"
+    - $(params.VOLUME_NAME)
+    - "--containerdiskname"
+    - $(params.CONTAINER_DISK_NAME)
+    - "--enablevirtsysprep"
+    - $(params.ENABLE_VIRT_SYSPREP)
+    resources:
```

Comment:
use computeResources instead (in v1, resources are deprecated)

Reply:
Done.